docs/AQL/grammar/AqlParser.g4

```diff
@@ -0,0 +1,214 @@
+//  copyright:    Copyright (c) 2020 openEHR Foundation
+//  license:      Apache 2.0 License <http://www.apache.org/licenses/LICENSE-2.0.html>
+//
+
+parser grammar AqlParser;
+
+options { tokenVocab=AqlLexer; }
+
+selectQuery
+    : selectClause fromClause whereClause? orderByClause? limitClause? MINUSMINUS? EOF
+    ;
+
+selectClause
+    : SELECT top? selectExpr (COMMA selectExpr)*
+    ;
+
+fromClause
+    : FROM fromExpr
+	;
+
+whereClause
+    : WHERE whereExpr
+    ;
+
+orderByClause
+    : ORDER BY orderByExpr (COMMA orderByExpr)*
+    ;
+
+limitClause
+    : LIMIT limit=NN_INTEGER (OFFSET offset=NN_INTEGER)?
+    ;
+
+
+selectExpr
+	: columnVar (AS IDENTIFIER)?
+	;
+
+fromExpr
+    : containsExpr
+    ;
+
+whereExpr
+    : NOT? identifiedExpr
+    | whereExpr AND whereExpr
+    | whereExpr OR whereExpr
+    | OPEN_PAR whereExpr CLOSE_PAR
+    ;
+
+orderByExpr
+	: identifiedPath order=(DESCENDING|DESC|ASCENDING|ASC)?
+	;
+
+
+columnVar
+	: identifiedPath
+	| aggregateFunctionCall
+	| functionCall
+	;
+
+containsExpr
+    : classExprOperand (CONTAINS containsExpr)?
+    | containsExpr AND containsExpr
+    | containsExpr OR containsExpr
+    | OPEN_PAR containsExpr CLOSE_PAR
+    ;
+
+identifiedExpr
+    : EXISTS identifiedPath
+    | identifiedPath COMPARISON_OPERATOR identifiedOperand
+    | functionCall COMPARISON_OPERATOR identifiedOperand
+    | identifiedPath LIKE likeOperand
+    | identifiedPath MATCHES OPEN_CURLY matchesOperand CLOSE_CURLY
+    ;
+
+identifiedOperand
+ 	: primitive
+ 	| PARAMETER
+ 	| identifiedPath
+ 	| functionCall
+ 	;
+
+identifiedPath
+    : IDENTIFIER pathPredicate? (SLASH objectPath)?
+    ;
+
+pathPredicate
+ 	: OPEN_BRACKET (standardPredicate | archetypePredicate | nodePredicate) CLOSE_BRACKET
+ 	;
+
+standardPredicate
+    : objectPath COMPARISON_OPERATOR pathPredicateOperand
+    ;
+
+archetypePredicate
+    : ARCHETYPEID
+    | PARAMETER
+    ;
+
+nodePredicate
+    : NODEID (COMMA (STRING|PARAMETER))?
+    | ARCHETYPEID (COMMA (STRING|PARAMETER))?
+    | PARAMETER
+    | objectPath COMPARISON_OPERATOR pathPredicateOperand
+    | objectPath MATCHES REGEXPATTERN
+    | nodePredicate AND nodePredicate
+    | nodePredicate OR nodePredicate
+    ;
+
+pathPredicateOperand
+    : primitive
+ 	| objectPath
+ 	| PARAMETER
+ 	;
+
+
+objectPath
+ 	: pathPart (SLASH pathPart)*
+ 	;
+pathPart
+ 	: IDENTIFIER pathPredicate?
+ 	;
+
+likeOperand
+    : STRING
+    | PARAMETER
+    ;
+matchesOperand
+ 	: valueListItem (COMMA valueListItem)*
+ 	| terminologyFunction
+ 	| URIVALUE
+ 	;
+valueListItem
+ 	: primitive
+ 	| PARAMETER
+ 	| terminologyFunction
+ 	;
+
+primitive
+    : STRING
+    | INTEGER
+    | FLOAT
+    | DOUBLE
+    | DATE
+    | BOOLEAN
+    ;
+
+
+
+classExprOperand
+	: IDENTIFIER IDENTIFIER? // RM_TYPE_NAME variable
+    | archetypedClassExpr
+    | versionedClassExpr
+	| versionClassExpr
```

Comment:
@teunvanhemert, should this also be simplified (by adding alt-names and avoiding all these extra rules) ?

Reply:
I'd say yes, the extra 'single-line' rules don't add that much.

Reply:
done, but it is now moved above around line 80